packages/embark-specialconfigs/src/index.js

```diff
@@ -273,4 +276,16 @@ class SpecialConfigs {
   }
 }
 
+function createDeployHookWithCustomLogContext(logPrefix, deployHook) {
+  return async function (dependencies) {
+    const originalLogFn = console.log;
+    console.log = function(...args) {
+      args.forEach(arg => originalLogFn(`${logPrefix} ${arg}`));
+    };
+    const result = await deployHook(dependencies);
```

Comment:
```suggestion
    let result;
    try {
      result = await deployHook(dependencies);
    } finally {
      // eslint-disable-next-line no-unsafe-finally
      console.log = originalLogFn;
    }
```
Should we ensure that `console.log` is restored even if `deployHook` blows up?

Reply:
Also, I'm wondering...

Isn't `console` a global object? Since `deployHook` is asynchronous and therefore concurrent with other code running inside the embark process, couldn't this result in completely unrelated code in the embark process using the substituted `.log` function if such code executes before `.log` is restored?

Reply:
Try running this in a node repl:

```js
const timer = setInterval(() => console.log(new Date()), 500);

setTimeout(async () => {
  const oldLog = console.log;
  console.log = () => oldLog('waffles');
  await (new Promise(res => { setTimeout(res, 2000); }));
  console.log = oldLog;
  setTimeout(() => process.exit(0), 1000);
}, 1000);
```

Reply:
Same effect if it's split up into modules like

```js
// file: a.js
const timer = setInterval(() => console.log(new Date()), 500);

// file b.js
setTimeout(async () => {
  const oldLog = console.log;
  console.log = () => oldLog('waffles');
  await (new Promise(res => { setTimeout(res, 2000); }));
  console.log = oldLog;
  setTimeout(() => process.exit(0), 1000);
}, 1000);
```

And then you do `node a.js`.

Reply:
Oh you have good points Michael. Not sure if there is a way to ensure that the context is correct

Reply:
@michaelsbradleyjr you're spot-on. That's why I mentioned that, overriding console.log temporarily feels a little dirty. Probably, chances are not super duper high that this will happen in practice (often), but high enough to raise this and definitely reconsider.

I guess, the better alternative would really be to have users use `logger` and add documentation, so they are aware of it.